common/src/main/java/com/wynntils/mc/mixin/ClientPacketListenerMixin.java

```diff
@@ -555,11 +559,26 @@ private void handleRemoveMobEffectPost(ClientboundRemoveMobEffectPacket packet,
 
     @Inject(
             method = "handleAddEntity(Lnet/minecraft/network/protocol/game/ClientboundAddEntityPacket;)V",
+            at =
+                    @At(
+                            value = "INVOKE",
+                            target =
+                                    "Lnet/minecraft/client/multiplayer/ClientPacketListener;postAddEntitySoundInstance(Lnet/minecraft/world/entity/Entity;)V",
+                            shift = At.Shift.AFTER))
+    private void handleAddEntity(ClientboundAddEntityPacket packet, CallbackInfo ci, @Local Entity entity) {
```

Comment:
For those wondering (and so you dont have to open your IDEA while reviewing), this is the same mixin, but we are not calling it at RETURN, but at the last instruction. Original code is like

```
if (entity != null) ...
else log error
```

we need the entity local to still exist, and also this fixes null entities posting this event. win-win I guess.